### ISSUE TITLE
Protocol.parseTargetHostAndSlot() is IPV4 oriented

### ISSUE THREAD

**Issue by allanwax** [xetorthio/jedis#982]
The code which splits the host part from the port part is very IPV4 oriented and thus it just splits on a `:`.  To handle IPV6 cases it should be something like:

``` java
...
int colon = messageInfo[2].lastIndexOf(":");
...
response[1] = messageInfo[2].substring(0, colon-1);
response[2] = messageInfo[2].substring(colon+1);
```

Since IPV6 uses colons to separate the parts and Jedis uses a colon to separate host from port, using the index of the last colon will select the host-port divider whether it's IPV4 or IPV6.

**Comment by marcosnils** [comment]
@allanwax I believe this should also be fixed here right:

https://github.com/xetorthio/jedis/blob/master/src/main/java/redis/clients/jedis/JedisSentinelPool.java#L123
https://github.com/xetorthio/jedis/blob/master/src/main/java/redis/clients/jedis/JedisSentinelPool.java#L169
https://github.com/xetorthio/jedis/blob/master/src/main/java/redis/clients/jedis/JedisSentinelPool.java#L178-L183
https://github.com/xetorthio/jedis/blob/master/src/main/java/redis/clients/util/ClusterNodeInformationParser.java#L37
https://github.com/xetorthio/jedis/blob/master/src/test/java/redis/clients/jedis/tests/utils/ClientKillerUtil.java#L10
https://github.com/xetorthio/jedis/blob/master/src/test/java/redis/clients/jedis/tests/HostAndPortUtil.java#L57

I'll submit a patch shortly

**Comment by allanwax** [comment]
Probably in more places than I know.  But yes, fix 'em all

**Comment by allanwax** [comment]
Sketchy code !!!  Without looking at the current HostAndPort !!!!!!

class HostAndPort {
...
private String host;
private String port;
...
public HostAndPort(String hostAndPort) {
// do proper splitting
host = ...;
port = ...;
}
...
public String getHost() {return host;}
public String getPort() {return port;}
...
}

**Comment by sazzad16** [comment]
Closed by #1342
